### ISSUE TITLE
Investigate Knative

### ISSUE THREAD

**Issue by digitalronin** [ministryofjustice/cloud-platform#818]
This is something Kerin [mentioned in Slack](https://mojdt.slack.com/archives/C514ETYJX/p1555060161005300)

[Knative](https://www.knative.dev/) is a tool for developing, building and deploying 'serverless' applications on kubernetes.

It forms the basis for [Google Cloud Run](https://cloud.google.com/run/)

We should investigate this, to see if it offers any significant benefit to our users. IMO, if the only benefits are more efficient utilisation of our servers, by not running containers until they receive traffic, that *isn't* worth worrying about. However, if it makes things simpler/easier/faster for our users, it might be worth trying it out.

Their [build component](https://www.knative.dev/docs/build/) might be useful.

Please check the slack thread above before starting this ticket, in case anyone has added an useful comments since I wrote this ticket.

**Comment by AntonyBishop** [comment]
Superseded by #1433
